src/clib/cool1d_multi_g.F

```diff
@@ -508,41 +514,41 @@ subroutine cool1d_multi_g(
 
          if (ispecies .gt. 1) then
             do i = is+1, ie+1
-               if ( itmask(i) ) then
+               if ( itmask(i) .ne. MASK_FALSE ) then
                   nH2 = 0.5_DKIND*(H2I(i,j,k) + H2II(i,j,k))
                   nother = (HeI(i,j,k) + HeII(i,j,k) +
      &                 HeIII(i,j,k))/4._DKIND +
      &                 HI(i,j,k) + HII(i,j,k) + de(i,j,k)
-#ifdef CALCULATE_TGAS_SELF_CONSISTENTLY
+
                   iter_tgas = 0
                   tgas_err = huge8
                   do while ((iter_tgas .lt. 100)
      &                 .and.(tgas_err .gt. 1.d-3))
-                  tgas0 = tgas(i)
-#endif
-                  if (nH2/nother .gt. 1.0e-3_DKIND) then
-                     x = 6100._DKIND/tgas(i) ! not quite self-consistent
-                     if (x .gt. 10._DKIND) then
-                        gamma2 = 0.5_DKIND*5._DKIND
+                     tgas0 = tgas(i)
+                     if (nH2/nother .gt. 1.0e-3_DKIND) then
+                        x = 6100._DKIND/tgas(i) ! not quite self-consistent
+                        if (x .gt. 10._DKIND) then
+                           gamma2 = 0.5_DKIND*5._DKIND
+                        else
+                           gamma2 = 0.5_DKIND*(5._DKIND + 2._DKIND*x**2
+     &                       * exp(x)/(exp(x)-1)**2)
+                        endif
                      else
-                        gamma2 = 0.5_DKIND*(5._DKIND + 2._DKIND*x**2 * 
-     &                       exp(x)/(exp(x)-1)**2)
+                        gamma2 = 2.5_DKIND
                      endif
-                  else
-                     gamma2 = 2.5_DKIND
-                  endif
-                  gamma2 = 1._DKIND + (nH2 + nother)/
-     &                 (nH2*gamma2 + nother/(gamma-1._DKIND))
+                     gamma2 = 1._DKIND + (nH2 + nother)/
+     &                    (nH2*gamma2 + nother/(gamma-1._DKIND))
 #ifdef CALCULATE_TGAS_SELF_CONSISTENTLY
-                  tgas(i) = max((gamma2 - 1._DKIND)*mmw(i)*e(i,j,k)*utem
-     &                        , temstart)
-                  tgas_err = dabs(tgas0 - tgas(i)) / tgas0
-                  iter_tgas = iter_tgas + 1
-                  end do
+                     tgas(i) = max((gamma2 - 1._DKIND)*mmw(i)*e(i,j,k)*
+     &                    utem, temstart)
+                     tgas_err = dabs(tgas0 - tgas(i)) / tgas0
+                     iter_tgas = iter_tgas + 1
 #else
-                  tgas(i) = tgas(i) * (gamma2 - 1._DKIND)/
-     &                 (gamma - 1._DKIND)
+                     tgas(i) = tgas(i) * (gamma2 - 1._DKIND)/
+     &                    (gamma - 1._DKIND)
+                     iter_tgas = 101
 #endif
+                  end do
```

Comment:
This is hard to parse just looking at the diff. Is this just fixing the tab level? Does the ifdef still cover this whole block?

Reply:
Originally the ifdef did **not** cover the whole block. 
- The ``do while`` and ``end do`` statements were conditionally included
- The change I made here was to always include the ``do while`` and ``end do`` statements
- This is needed for the automatic transcription logic

I have attempted to show what the code looked like before and after the change (hopefully I didn't mess up my copy/pasting from the diff).

Here is the original

```Fortran
#ifdef CALCULATE_TGAS_SELF_CONSISTENTLY
                  iter_tgas = 0
                  tgas_err = huge8
                  do while ((iter_tgas .lt. 100)
     &                 .and.(tgas_err .gt. 1.d-3))
                  tgas0 = tgas(i)
#endif
                  if (nH2/nother .gt. 1.0e-3_DKIND) then
                     x = 6100._DKIND/tgas(i) ! not quite self-consistent
                     if (x .gt. 10._DKIND) then
                        gamma2 = 0.5_DKIND*5._DKIND
                     else
                        gamma2 = 0.5_DKIND*(5._DKIND + 2._DKIND*x**2 * 
     &                       exp(x)/(exp(x)-1)**2)
                     endif
                  else
                     gamma2 = 2.5_DKIND
                  endif
                  gamma2 = 1._DKIND + (nH2 + nother)/
     &                 (nH2*gamma2 + nother/(gamma-1._DKIND))
ifdef CALCULATE_TGAS_SELF_CONSISTENTLY
                  tgas(i) = max((gamma2 - 1._DKIND)*mmw(i)*e(i,j,k)*utem
     &                        , temstart)
                  tgas_err = dabs(tgas0 - tgas(i)) / tgas0
                  iter_tgas = iter_tgas + 1
                  end do
#else
                  tgas(i) = tgas(i) * (gamma2 - 1._DKIND)/
     &                 (gamma - 1._DKIND)
#endif
```

After the changes, the block now looks like:
```FORTRAN
                  iter_tgas = 0
                  tgas_err = huge8
                  do while ((iter_tgas .lt. 100)
     &                 .and.(tgas_err .gt. 1.d-3))
                     tgas0 = tgas(i)
                     if (nH2/nother .gt. 1.0e-3_DKIND) then
                        x = 6100._DKIND/tgas(i) ! not quite self-consistent
                        if (x .gt. 10._DKIND) then
                           gamma2 = 0.5_DKIND*5._DKIND
                        else
                           gamma2 = 0.5_DKIND*(5._DKIND + 2._DKIND*x**2
     &                       * exp(x)/(exp(x)-1)**2)
                        endif
                     else
                        gamma2 = 2.5_DKIND
                     endif
                     gamma2 = 1._DKIND + (nH2 + nother)/
     &                    (nH2*gamma2 + nother/(gamma-1._DKIND))
#ifdef CALCULATE_TGAS_SELF_CONSISTENTLY
                     tgas(i) = max((gamma2 - 1._DKIND)*mmw(i)*e(i,j,k)*
     &                    utem, temstart)
                     tgas_err = dabs(tgas0 - tgas(i)) / tgas0
                     iter_tgas = iter_tgas + 1
#else
                     tgas(i) = tgas(i) * (gamma2 - 1._DKIND)/
     &                    (gamma - 1._DKIND)
                     iter_tgas = 101
#endif
                  end do
```

Reply:
Thanks, I was concerned that the inner bit might have been needed elsewhere, but it's only populating the scalar gamma2, so clearly not. This is fine. Probably something else to refactor or turn into a runtime parameter.